src/audio/kpb.c

```diff
@@ -861,6 +863,26 @@ static uint64_t kpb_draining_task(void *arg)
 	return 0;
 }
 
+void kpb_drain_data(void *source, struct comp_buffer *sink, size_t size)
+{
+	int16_t *dest;
+	int16_t *src = (int16_t *)source;
+	uint32_t i;
+	uint32_t j = 0;
+	uint32_t channel;
+	uint32_t frames = (size / ((KPB_SAMPLE_CONTAINER_SIZE/8) *
+			   KPB_NR_OF_CHANNELS));
+
+	for (i = 0; i < frames; i++) {
+		for (channel = 0; channel < KPB_NR_OF_CHANNELS; channel++) {
+			dest = buffer_write_frag_s16(sink, j);
```

Comment:
How shall we handle it once we are running with 24bit captured data? better to get it from fmt?

Reply:
The support for more sampling options will be provided as separate patch.